scripts/ci/build-site.py

```
@@ -0,0 +1,200 @@
+    archive = work / f"{line}.tar.gz"
+    if not fetch(f"{site}/data/{line}/results.tar.gz", archive):
+        return None
+    unpacked = work / line / "results"
+    unpacked.mkdir(parents=True, exist_ok=True)
+    with tarfile.open(archive, "r:gz") as bundle:
+        root = unpacked.resolve()
+        for member in bundle.getmembers():
+            # Архив приезжает с сайта, а не из дерева: наружу его не пускаем.
+            if not str((unpacked / member.name).resolve()).startswith(str(root)):
+                raise SystemExit(f"архив линии {line} ведёт наружу: {member.name}")
+        bundle.extractall(unpacked)
```

Ruff (0.16.3)

[error] Uses of `tarfile.extractall()`

(S202)